apps/mobile/src/screens/Onboarding/OnboardingRecommendedUsers.tsx

```diff
@@ -0,0 +1,201 @@
+      @refetchable(queryName: "OnboardingRecommendedUsersInnerQuery") {
+        viewer {
+          ... on Viewer {
+            id
+
+            suggestedUsers(last: $usersLast, before: $usersBefore)
+              @connection(key: "OnboardingRecommendedUsersInnerFragment_suggestedUsers") {
+              edges {
+                node {
+                  id
+                  __typename
+                  ...UserFollowListFragment
+                }
+              }
+            }
+          }
+        }
+        ...useFollowAllRecommendedUsersQueryFragment
+        ...UserFollowListQueryFragment
+      }
+    `,
+    queryRef
+  );
+
+  const navigation = useNavigation<LoginStackNavigatorProp>();
+
+  const user = followingPagination?.viewer?.id;
+
+  const suggestedFollowingIds = useMemo(() => {
+    const userIds = [];
+
+    for (const edge of followingPagination.viewer?.suggestedUsers?.edges ?? []) {
+      if (edge?.node?.__typename === 'GalleryUser') {
+        userIds.push(edge?.node.id);
+      }
+    }
+
+    return userIds;
+  }, [followingPagination]);
+
+  const windowDimensions = useWindowDimensions();
+
+  const heightOfUserList = windowDimensions.height - HEIGHT_OF_FIXED_ITEMS_ON_SCREEN;
+
+  const followAllRecommendedUsers = useFollowAllRecommendedUsers({
+    suggestedFollowingIds: suggestedFollowingIds,
+    queryRef: followingPagination,
+  });
+
+  const handleNext = useCallback(async () => {
+    if (!user) return null;
+    await navigateToNotificationUpsellOrHomeScreen(navigation, true);
+  }, [navigation, user]);
+
+  const handleFollowAll = useCallback(async () => {
+    if (!user) return null;
+    followAllRecommendedUsers();
+    setTimeout(async () => await handleNext(), 800);
+  }, [user, followAllRecommendedUsers, handleNext]);
+
+  const handleBack = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  const { top } = useSafeAreaInsets();
+
+  const handleLoadMore = useCallback(() => {
+    if (hasPrevious) {
+      loadPrevious(USERS_PER_PAGE);
+    }
+  }, [hasPrevious, loadPrevious]);
+
+  const recommendedUsers = useMemo(() => {
+    const list = followingPagination.viewer?.suggestedUsers?.edges?.map((edge) => edge?.node) ?? [];
+    return removeNullValues(list);
+  }, [followingPagination.viewer?.suggestedUsers?.edges]);
+
+  const [hasFollowedSomeone, setHasFollowedSomeone] = useState(false);
+
+  return (
+    <View style={{ paddingTop: top }} className="bg-white">
+      <View className="flex flex-col flex-grow space-y-4 px-4 bg-white">
+        <View className="relative flex-row items-center justify-between ">
+          <BackButton onPress={handleBack} />
+          <GalleryTouchableOpacity
+            onPress={handleNext}
+            className="flex flex-row items-center space-x-2"
+            eventElementId="Next button on onboarding recommended users screen"
+            eventName="Next button on onboarding recommended users screen"
```

Comment:
I think it will be great for data if we track this state. 

if the user skip - event name would be `Skip button on .....`
if the user follow someone - `Next button on onboarding recommended users screen`

Reply:
righttt. I added a variant now to track this? do you think that works or would the conditional string be better here?

Reply:
variant is perfect!